### PR TITLE
Update to conversion tool to work in current versions

### LINKED ISSUE #6104: rename influx_tsm in 0.11+
In InfluxDB 0.10, the `influx_tsm` binary was capable of converting `b1`/`bz1` shards to `tsm` shards, in addition to the other shard inspection functions.

In InfluxDB 0.11+ the `influx_tsm` binary is no longer capable of converting shards.

This is very likely to lead to user confusion during upgrades from 0.9 to 0.11+, as all upgrades from 0.9 reference the tool as the way to convert the existing shards to the `tsm` format.

To mitigate that, can we rename `influx_tsm` in 0.11+?

### PR DESCRIPTION
**Pull request by joelegasse**
Fixes #6104, #6135, #6100, #6092, and probably others

After adding type-switches to the tsm1 packages, the custom
implementation found in the conversion tool broke. This change uses
tsm1.NewValue() instead of a custom implementation.

This change also ensures that the tsm1.Value interface can only be
implemented internally to allow for the optimized type-switch based
encoding

### CODE REVIEW

File: tsdb/engine/tsm1/encoding.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ type Value interface {
 	Value() interface{}
 	Size() int
 	String() string
+
+	internalOnly()
```
Comment (jwilder) [inline tsdb/engine/tsm1/encoding.go]: What is this for?

Comment (jwilder) [inline tsdb/engine/tsm1/encoding.go]: To prevent using custom `Value` implementations?

Comment (joelegasse) [inline tsdb/engine/tsm1/encoding.go]: Yes, it makes sure that all `tsm1.Value`s are created by this package.

**Comment by jwilder** [comment]
:+1: on green.

**Comment by jwilder** [comment]
Oh.  Needs changelog update.

**Pull request merged**